scvi/inference/posterior.py

```diff
@@ -1144,9 +1144,31 @@ def get_stats(self):
         return libraries.ravel()
 
     @torch.no_grad()
-    def get_sample_scale(self, transform_batch=None):
+    def get_sample_scale(self, transform_batch=None, gene=None, library_size=1, batch_size=128):
+        ''' Returns the frequencies of expression for the data.
+
+        :param transform_batch: Batches to condition on.
+        If transform_batch is:
+            - None, then real observed batch is used
+            - int, then batch transform_batch is used
+            - list of int, then px_rates are averaged over provided batches.
+        :param gene: (optional) Return frequencies of expression for a single gene. This can
+            save a lot of memory when working with large datasets and only a single gene is
+            of interest. (default=None)
+        :param library_size: (optional) Scale the expression frequencies to a common library size.
+            This allows gene expression levels to be interpreted on a common scale of relevant
+            magnitude (default=1, no scaling).
+        :param batch_size: Size of minibatches of the data to use. When only accessing a single
+            gene a larger batch size (e.g. 10000) substantially reduces the time spent on GPU
+            data transfer (default=128).        
+        '''
+        if gene is None:
+            gene_idx = slice(None)
+        else:
+            gene_idx = [np.where(self.gene_dataset.gene_names == gene)[0][0]]
```

Comment:
Do you think it's helpful if we allow gene to be a subset of genes?

Reply:
I was thinking about that. But I wasn't sure what a good way would be to relate which returned columns to which genes. (Related, there should probably be a check here to make sure 1) the gene is in the gene list, 2) the gene name is unique).

Reply:
So we have this function....
https://github.com/YosefLab/scVI/blob/4a5eef104fdd1c5eb548d7ab85704ba99f16a00f/scvi/dataset/dataset.py#L896-L900

We can use that `subset_genes` returned by this to subset the scale, and also to help define the return value, which could be a pandas dataframe in this case. This makes it easy to ignore the input order.

Reply:
and then it also doesn't matter if it's one gene or many genes. the input would be a list of genes.

Reply:
I think I arrived at a decent compromise: by default just return an array, but can specify to return a `DataFrame` with correct column names.